test/parallel/test-timers-api-refs.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+const common = require('../common');
+const timers = require('timers');
+
+// delete global APIs to make sure they're not relied on by the internal timers
+// code
+delete global.setTimeout;
+delete global.clearTimeout;
+delete global.setInterval;
+delete global.clearInterval;
+delete global.setImmediate;
+delete global.clearImmediate;
```

Comment:
I agree that if you override `timers.active()`, then you're on your own and have to manage any possible side effects yourself. (That said, I do think it's worthwhile for `timers.js` to guard against the problem internally, as it does in this PR, as long as there's no reason to suspect a performance hit or other unanticipated side effect.)